src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -5311,6 +5311,10 @@ public void assembleSig(Type type) {
                     }
                     append('T');
                     append(type.tsym.name);
+                    NullMarker nullMarker = type.getNullMarker();
```

Comment:
Better to use the predicates in `Types` which take into account the command line options (e.g. whether to treat type-variables as `*` or not)